<center><b><h1>STUDENT PERFORMANCE ANALYSIS</b></h1>

<img src="std p.png" width = 1000 height = 30>

<b><h2><center>DESCRIPTION</center></b></h2>

This project aims to analyze the factors that influence student's academic performance and identify key patterns in their study habits and lifestyle. It also focuses on predicting exam scores and grades using behavioral and demographic factors.<p>
This dataset contains information on 1000 students, including study habits (hours studied, attendance), lifestyle factors (sleep, extracurricular activities, part-time job), demographic details (gender, parental education), and academic outcomes (final exam score and letter grade). It's ideal for exploratory data analysis, regression, and classification tasks to predict student performance based on behavioral and demographic factors. Use cases: Predicting exam scores, grade classification, EDA on academic performance factors, feature importance analysis.

<center><h3>BASIC ANALYSIS</center></h3>

In [4]:
import pandas as pd
import warnings
warnings.filterwarnings("ignore")

In [5]:
df = pd.read_csv("student_performance_dataset.csv")

In [6]:
df

,student_id,gender,study_time_hours,attendance_percent,sleep_hours,parental_education,internet_access,extracurricular_activities,part_time_job,previous_grade,final_exam_score,final_grade
0,1,Male,4.0,98.0,6.5,Bachelors,Yes,Yes,No,76.9,100.0,A
1,2,Female,6.3,100.0,5.7,High School,Yes,Yes,Yes,75.5,100.0,A
2,3,Male,4.9,85.3,7.9,Bachelors,Yes,No,Yes,88.5,97.3,A
3,4,Male,2.6,77.5,8.0,NaN,Yes,Yes,No,85.1,83.8,B
4,5,Male,2.2,89.6,4.6,Bachelors,Yes,No,Yes,61.8,68.3,D
...,...,...,...,...,...,...,...,...,...,...,...,...
995,996,Male,2.8,75.4,8.2,Masters,Yes,Yes,No,60.5,70.7,C
996,997,Male,6.7,88.4,7.1,NaN,No,Yes,No,82.2,99.5,A
997,998,Female,2.6,84.5,8.0,High School,Yes,Yes,Yes,65.2,79.2,C
998,999,Female,4.6,85.3,8.1,High School,No,No,Yes,52.2,82.2,B


In [7]:
df.shape

(1000, 12)

In [8]:
df.columns

Index(['student_id', 'gender', 'study_time_hours', 'attendance_percent',
       'sleep_hours', 'parental_education', 'internet_access',
       'extracurricular_activities', 'part_time_job', 'previous_grade',
       'final_exam_score', 'final_grade'],
      dtype='str')

In [9]:
df.dtypes

student_id                      int64
gender                            str
study_time_hours              float64
attendance_percent            float64
sleep_hours                   float64
parental_education                str
internet_access                   str
extracurricular_activities        str
part_time_job                     str
previous_grade                float64
final_exam_score              float64
final_grade                       str
dtype: object

In [10]:
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 12 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   student_id                  1000 non-null   int64  
 1   gender                      1000 non-null   str    
 2   study_time_hours            1000 non-null   float64
 3   attendance_percent          1000 non-null   float64
 4   sleep_hours                 1000 non-null   float64
 5   parental_education          898 non-null    str    
 6   internet_access             1000 non-null   str    
 7   extracurricular_activities  1000 non-null   str    
 8   part_time_job               1000 non-null   str    
 9   previous_grade              1000 non-null   float64
 10  final_exam_score            1000 non-null   float64
 11  final_grade                 1000 non-null   str    
dtypes: float64(5), int64(1), str(6)
memory usage: 93.9 KB


In [11]:
df.describe()

,student_id,study_time_hours,attendance_percent,sleep_hours,previous_grade,final_exam_score
count,1000.000000,1000.000000,1000.000000,1000.000000,1000.000000,1000.000000
mean,500.500000,3.570700,85.092300,6.799500,69.740900,83.543500
std,288.819436,1.478559,9.270685,1.203527,12.613425,10.341333
min,1.000000,0.500000,54.800000,3.200000,31.300000,46.800000
25%,250.750000,2.600000,78.800000,5.900000,61.000000,76.075000
50%,500.500000,3.600000,85.200000,6.800000,69.600000,83.800000
75%,750.250000,4.500000,91.900000,7.600000,78.400000,91.525000
max,1000.000000,8.100000,100.000000,10.000000,100.000000,100.000000


<h3><center>OBSERVATION</center></h3>

In [12]:
df.isnull().sum()

student_id                      0
gender                          0
study_time_hours                0
attendance_percent              0
sleep_hours                     0
parental_education            102
internet_access                 0
extracurricular_activities      0
part_time_job                   0
previous_grade                  0
final_exam_score                0
final_grade                     0
dtype: int64

In [13]:
df.isnull().sum().sum()

np.int64(102)

<h3>INSIGHT :</h3>  Only the parental_education column has missing values. It contains 102 null values. All other columns have complete data.

In [14]:
df.duplicated().sum()

np.int64(0)

In [15]:
df["parental_education"]=df["parental_education"].fillna(df["parental_education"].mode()[0],inplace=True)

In [16]:
df

,student_id,gender,study_time_hours,attendance_percent,sleep_hours,parental_education,internet_access,extracurricular_activities,part_time_job,previous_grade,final_exam_score,final_grade
0,1,Male,4.0,98.0,6.5,Bachelors,Yes,Yes,No,76.9,100.0,A
1,2,Female,6.3,100.0,5.7,High School,Yes,Yes,Yes,75.5,100.0,A
2,3,Male,4.9,85.3,7.9,Bachelors,Yes,No,Yes,88.5,97.3,A
3,4,Male,2.6,77.5,8.0,High School,Yes,Yes,No,85.1,83.8,B
4,5,Male,2.2,89.6,4.6,Bachelors,Yes,No,Yes,61.8,68.3,D
...,...,...,...,...,...,...,...,...,...,...,...,...
995,996,Male,2.8,75.4,8.2,Masters,Yes,Yes,No,60.5,70.7,C
996,997,Male,6.7,88.4,7.1,High School,No,Yes,No,82.2,99.5,A
997,998,Female,2.6,84.5,8.0,High School,Yes,Yes,Yes,65.2,79.2,C
998,999,Female,4.6,85.3,8.1,High School,No,No,Yes,52.2,82.2,B


<h3>INSIGHT :</h3>  The missing values in the parental_education column were replaced with the most frequent education level. This keeps all records in the dataset and using the mode is a suitable method for filling missing categorical values.

In [17]:
df.isnull().sum()

student_id                    0
gender                        0
study_time_hours              0
attendance_percent            0
sleep_hours                   0
parental_education            0
internet_access               0
extracurricular_activities    0
part_time_job                 0
previous_grade                0
final_exam_score              0
final_grade                   0
dtype: int64

So the dataset now has no missing data and is ready for analysis.

In [18]:
df["gender"].value_counts()

gender
Female    510
Male      490
Name: count, dtype: int64

<h3>INSIGHT :</h3> The dataset has 510 female and 490 male students, so the number of students is almost equally distributed by gender.

In [19]:
df["internet_access"].value_counts()

internet_access
Yes    854
No     146
Name: count, dtype: int64

<h3>INSIGHT :</h3> 854 students have internet access. only 146 students do not internet access. So most students can access online learning resources.

In [20]:
df["extracurricular_activities"].value_counts()

extracurricular_activities
Yes    572
No     428
Name: count, dtype: int64

<h3>INSIGHT :</h3> 572 students participate in extracurricular activities. 428 students not participate . So more students active in the extracurricular activities.

In [21]:
df["part_time_job"].value_counts()

part_time_job
No     684
Yes    316
Name: count, dtype: int64

<h3>INSIGHT :</h3> 316 students have a part-time job. The majority 684 students do not work. So most students focus mainly on their studies.

In [22]:
df.groupby("gender")["attendance_percent"].mean()

gender
Female    84.963333
Male      85.226531
Name: attendance_percent, dtype: float64

<h3> INSIGHT : </h3>  This analysis compares the average attendance of female and male students. Male students have an average attendance of 85.23%, while female students have 84.96%. 

In [23]:
df.groupby("gender")["final_exam_score"].mean()

gender
Female    83.606863
Male      83.477551
Name: final_exam_score, dtype: float64

<h3> INSIGHT : </h3> This analysis compares the average final exam score of male and female students. Female students have an average score of 83.61, while male students have 83.48.

In [24]:
df[df["final_exam_score"]==df["final_exam_score"].max()]

,student_id,gender,study_time_hours,attendance_percent,sleep_hours,parental_education,internet_access,extracurricular_activities,part_time_job,previous_grade,final_exam_score,final_grade
0,1,Male,4.0,98.0,6.5,Bachelors,Yes,Yes,No,76.9,100.0,A
1,2,Female,6.3,100.0,5.7,High School,Yes,Yes,Yes,75.5,100.0,A
21,22,Male,3.3,100.0,7.8,High School,Yes,Yes,No,83.9,100.0,A
32,33,Female,5.3,100.0,7.3,Masters,Yes,Yes,No,67.7,100.0,A
34,35,Female,6.6,90.4,8.1,Masters,Yes,Yes,Yes,69.7,100.0,A
...,...,...,...,...,...,...,...,...,...,...,...,...
934,935,Female,3.6,89.4,7.0,Bachelors,Yes,No,No,95.3,100.0,A
948,949,Female,3.7,93.9,8.5,High School,No,No,No,93.0,100.0,A
954,955,Male,3.7,92.0,7.6,Masters,Yes,Yes,No,65.9,100.0,A
967,968,Female,4.6,71.3,6.9,Bachelors,Yes,Yes,No,95.5,100.0,A


<h3> INSIGHT : </h3>  This analysis identifies the students with the highest final exam score. A total of 78 students achieved maximum score of 100. These students are the top performer in the dataset.

In [25]:
df[df["final_exam_score"]==df["final_exam_score"].min()]

,student_id,gender,study_time_hours,attendance_percent,sleep_hours,parental_education,internet_access,extracurricular_activities,part_time_job,previous_grade,final_exam_score,final_grade
147,148,Male,1.7,83.5,5.7,Masters,Yes,Yes,Yes,59.2,46.8,F


<h3> INSIGHT : </h3> This displays the student(s) with the lowest final exam score. It helps identify the lowest academic performer in the dataset.

In [26]:
df.groupby("parental_education")["final_exam_score"].mean().sort_values(ascending=False)

parental_education
PhD            85.062000
Masters        84.034239
Bachelors      83.400325
High School    83.276856
Name: final_exam_score, dtype: float64

<h3> INSIGHT : </h3>  This analysis compares the average final exam score based on parental education. Students whose parents have a PhD have the highest average score, while students whose parents have a High School education have the lowest average score. 

In [27]:
df.groupby("internet_access")["final_exam_score"].mean()

internet_access
No     79.258219
Yes    84.276112
Name: final_exam_score, dtype: float64

<h3> INSIGHT : </h3>  This analysis compares the average final exam score based on internet access. Students with internet have a slightly higher average score than those without it. This suggest that internet access may support better learning opportunities. 

In [28]:
df.groupby("extracurricular_activities")["final_exam_score"].mean()

extracurricular_activities
No     83.284346
Yes    83.737413
Name: final_exam_score, dtype: float64

<h3> INSIGHT : </h3>  This compares the average exam score based on extracurricular activities. Students who participate scored 83.74, while those who do not scored 83.28. The difference is very small, so extracurricular activities shows little effect on average exam performance. 

In [29]:
df.groupby("part_time_job")["final_exam_score"].mean()

part_time_job
No     84.582310
Yes    81.294937
Name: final_exam_score, dtype: float64

<h3> INSIGHT : </h3>  This compares the average exam score of students with and without part-time job. Students without part-time job scored 84.58, while students with part-time job scored 81.29. This suggest that students without part-time jobs performed slightly better on average.

In [30]:
df.groupby("study_time_hours")["final_exam_score"].mean()

study_time_hours
0.5     71.026087
0.7     71.825000
0.8     75.633333
0.9     71.433333
1.0     75.587500
          ...    
7.1    100.000000
7.2     97.475000
7.3     93.900000
7.4     99.133333
8.1     98.300000
Name: final_exam_score, Length: 70, dtype: float64

<h3> INSIGHT : </h3>  This analysis groups students by their study time and calculates the average final exam score. The output shows that higher study time is generally associated with higher exam scores. This suggest that regular study has a positive impact on academic performance.

In [31]:
df.groupby("sleep_hours")["final_exam_score"].mean()

sleep_hours
3.2      83.90
3.3      83.90
3.4      71.40
3.5      61.80
3.6      92.90
         ...  
9.5      86.45
9.6      67.80
9.7      81.85
9.9     100.00
10.0     90.00
Name: final_exam_score, Length: 65, dtype: float64

<h3> INSIGHT : </h3>  This analysis groups students by their sleep hours and calculates the average final exam score. The results show that academic performance varies across different sleep durations. Students with around 7-9 hours of sleep generally achieve higher average scores than those with very low sleep.

In [32]:
df.nlargest(5,"final_exam_score")

,student_id,gender,study_time_hours,attendance_percent,sleep_hours,parental_education,internet_access,extracurricular_activities,part_time_job,previous_grade,final_exam_score,final_grade
0,1,Male,4.0,98.0,6.5,Bachelors,Yes,Yes,No,76.9,100.0,A
1,2,Female,6.3,100.0,5.7,High School,Yes,Yes,Yes,75.5,100.0,A
21,22,Male,3.3,100.0,7.8,High School,Yes,Yes,No,83.9,100.0,A
32,33,Female,5.3,100.0,7.3,Masters,Yes,Yes,No,67.7,100.0,A
34,35,Female,6.6,90.4,8.1,Masters,Yes,Yes,Yes,69.7,100.0,A


<h3> INSIGHT : </h3>  This displays the top 5 students with the highest final exam scores. It helps identify the best academic performers in the dataset.

In [33]:
df.nsmallest(5,"final_exam_score")

,student_id,gender,study_time_hours,attendance_percent,sleep_hours,parental_education,internet_access,extracurricular_activities,part_time_job,previous_grade,final_exam_score,final_grade
147,148,Male,1.7,83.5,5.7,Masters,Yes,Yes,Yes,59.2,46.8,F
760,761,Female,1.9,69.0,6.0,High School,Yes,No,Yes,34.5,53.2,F
715,716,Female,0.5,76.0,7.8,High School,No,Yes,No,62.5,54.1,F
539,540,Female,1.2,75.8,7.1,Bachelors,Yes,Yes,Yes,45.0,57.3,F
364,365,Male,2.1,58.0,8.9,Bachelors,Yes,Yes,No,61.5,57.4,F


<h3> INSIGHT : </h3> This displays the bottom 5 students with lowest final exam scores. It helps identify students who may need additional academic support.

In [34]:
df["score_difference"]=df["final_exam_score"]-df["previous_grade"]

In [35]:
df

,student_id,gender,study_time_hours,attendance_percent,sleep_hours,parental_education,internet_access,extracurricular_activities,part_time_job,previous_grade,final_exam_score,final_grade,score_difference
0,1,Male,4.0,98.0,6.5,Bachelors,Yes,Yes,No,76.9,100.0,A,23.1
1,2,Female,6.3,100.0,5.7,High School,Yes,Yes,Yes,75.5,100.0,A,24.5
2,3,Male,4.9,85.3,7.9,Bachelors,Yes,No,Yes,88.5,97.3,A,8.8
3,4,Male,2.6,77.5,8.0,High School,Yes,Yes,No,85.1,83.8,B,-1.3
4,5,Male,2.2,89.6,4.6,Bachelors,Yes,No,Yes,61.8,68.3,D,6.5
...,...,...,...,...,...,...,...,...,...,...,...,...,...
995,996,Male,2.8,75.4,8.2,Masters,Yes,Yes,No,60.5,70.7,C,10.2
996,997,Male,6.7,88.4,7.1,High School,No,Yes,No,82.2,99.5,A,17.3
997,998,Female,2.6,84.5,8.0,High School,Yes,Yes,Yes,65.2,79.2,C,14.0
998,999,Female,4.6,85.3,8.1,High School,No,No,Yes,52.2,82.2,B,30.0


<h3>INSIGHT : </h3>  A new column called score_difference was created. It shows how much each student's final exam score changed compared to their previous grade.

In [36]:
df[["student_id","previous_grade","final_exam_score","score_difference"]].head(20)

,student_id,previous_grade,final_exam_score,score_difference
0,1,76.9,100.0,23.1
1,2,75.5,100.0,24.5
2,3,88.5,97.3,8.8
3,4,85.1,83.8,-1.3
4,5,61.8,68.3,6.5
5,6,79.8,81.6,1.8
6,7,58.6,69.6,11.0
7,8,78.4,88.0,9.6
8,9,63.8,84.4,20.6
9,10,57.7,74.3,16.6


<h3> INSIGHT : </h3>  This displays the first 20 students with their previous grade, final exam score, and score difference. It helps compare each student's academic improvement or decline.

In [37]:
df[df["score_difference"]>0]

,student_id,gender,study_time_hours,attendance_percent,sleep_hours,parental_education,internet_access,extracurricular_activities,part_time_job,previous_grade,final_exam_score,final_grade,score_difference
0,1,Male,4.0,98.0,6.5,Bachelors,Yes,Yes,No,76.9,100.0,A,23.1
1,2,Female,6.3,100.0,5.7,High School,Yes,Yes,Yes,75.5,100.0,A,24.5
2,3,Male,4.9,85.3,7.9,Bachelors,Yes,No,Yes,88.5,97.3,A,8.8
4,5,Male,2.2,89.6,4.6,Bachelors,Yes,No,Yes,61.8,68.3,D,6.5
5,6,Female,4.2,78.2,5.7,High School,Yes,No,No,79.8,81.6,B,1.8
...,...,...,...,...,...,...,...,...,...,...,...,...,...
995,996,Male,2.8,75.4,8.2,Masters,Yes,Yes,No,60.5,70.7,C,10.2
996,997,Male,6.7,88.4,7.1,High School,No,Yes,No,82.2,99.5,A,17.3
997,998,Female,2.6,84.5,8.0,High School,Yes,Yes,Yes,65.2,79.2,C,14.0
998,999,Female,4.6,85.3,8.1,High School,No,No,Yes,52.2,82.2,B,30.0


<h3> INSIGHT : </h3> This filters students with a positive score difference. It shows all students who scored higher in final exam than in their previous grade, indicating academic improvement. 

In [38]:
df.rename(columns={"study_time_hours":"study_hours","attendance_percent":"attendance"}, inplace=True)

In [39]:
df.columns

Index(['student_id', 'gender', 'study_hours', 'attendance', 'sleep_hours',
       'parental_education', 'internet_access', 'extracurricular_activities',
       'part_time_job', 'previous_grade', 'final_exam_score', 'final_grade',
       'score_difference'],
      dtype='str')

<h3>INSIGHT :</h3> Long column names were renamed into shorter names. This makes the code easier to read. It also improves data handling.

In [40]:
df[df["study_hours"]>5]

,student_id,gender,study_hours,attendance,sleep_hours,parental_education,internet_access,extracurricular_activities,part_time_job,previous_grade,final_exam_score,final_grade,score_difference
1,2,Female,6.3,100.0,5.7,High School,Yes,Yes,Yes,75.5,100.0,A,24.5
7,8,Male,6.2,86.4,6.0,High School,Yes,Yes,No,78.4,88.0,B,9.6
8,9,Male,5.3,81.3,6.7,Bachelors,Yes,No,No,63.8,84.4,B,20.6
11,12,Male,5.5,75.3,5.6,High School,Yes,Yes,No,37.9,76.7,C,38.8
13,14,Male,5.4,78.4,6.9,Bachelors,No,No,No,67.9,93.3,A,25.4
...,...,...,...,...,...,...,...,...,...,...,...,...,...
985,986,Female,6.1,79.5,5.4,High School,Yes,No,No,75.6,96.4,A,20.8
988,989,Female,5.7,96.2,4.8,High School,Yes,No,No,41.5,80.9,B,39.4
991,992,Female,5.1,71.7,5.6,Bachelors,Yes,Yes,No,51.0,80.4,B,29.4
993,994,Female,6.2,87.6,5.9,PhD,Yes,No,No,75.4,92.7,A,17.3


<h3>INSIGHT : </h3>  166 students study more than 5 hours daily. These students can be analyzed separately. Their performance may be higher.

In [41]:
df[df["attendance"]>90]

,student_id,gender,study_hours,attendance,sleep_hours,parental_education,internet_access,extracurricular_activities,part_time_job,previous_grade,final_exam_score,final_grade,score_difference
0,1,Male,4.0,98.0,6.5,Bachelors,Yes,Yes,No,76.9,100.0,A,23.1
1,2,Female,6.3,100.0,5.7,High School,Yes,Yes,Yes,75.5,100.0,A,24.5
6,7,Male,1.5,100.0,5.0,High School,Yes,Yes,Yes,58.6,69.6,D,11.0
12,13,Male,3.3,97.0,7.0,Masters,Yes,Yes,Yes,49.0,82.6,B,33.6
15,16,Male,2.6,90.4,7.0,High School,Yes,Yes,No,65.6,84.8,B,19.2
...,...,...,...,...,...,...,...,...,...,...,...,...,...
977,978,Female,2.0,96.1,5.2,Bachelors,Yes,Yes,No,60.7,72.1,C,11.4
979,980,Female,3.3,100.0,9.1,High School,Yes,Yes,No,64.0,93.9,A,29.9
983,984,Male,2.2,97.4,7.0,Masters,Yes,Yes,No,64.4,85.0,B,20.6
988,989,Female,5.7,96.2,4.8,High School,Yes,No,No,41.5,80.9,B,39.4


<h3>INSIGHT :</h3> 319 students have attendance above 90%. They represent students with regular attendance. It may influence exam performance.

In [42]:
df.sort_values(by="final_exam_score",ascending=False)

,student_id,gender,study_hours,attendance,sleep_hours,parental_education,internet_access,extracurricular_activities,part_time_job,previous_grade,final_exam_score,final_grade,score_difference
381,382,Female,6.2,83.2,7.3,PhD,Yes,No,No,86.1,100.0,A,13.9
383,384,Male,6.2,90.5,6.6,High School,Yes,Yes,No,84.0,100.0,A,16.0
387,388,Male,5.0,86.9,4.9,High School,Yes,Yes,No,100.0,100.0,A,0.0
312,313,Female,4.9,90.1,7.5,High School,Yes,Yes,No,92.5,100.0,A,7.5
314,315,Male,2.7,88.5,7.3,Masters,Yes,Yes,No,99.3,100.0,A,0.7
...,...,...,...,...,...,...,...,...,...,...,...,...,...
364,365,Male,2.1,58.0,8.9,Bachelors,Yes,Yes,No,61.5,57.4,F,-4.1
539,540,Female,1.2,75.8,7.1,Bachelors,Yes,Yes,Yes,45.0,57.3,F,12.3
715,716,Female,0.5,76.0,7.8,High School,No,Yes,No,62.5,54.1,F,-8.4
760,761,Female,1.9,69.0,6.0,High School,Yes,No,Yes,34.5,53.2,F,18.7


<h3>INSIGHT :</h3> The highest score is 100 and the lowest is 46.8. Sorting identifies top and low performers. It helps in ranking students.

In [43]:
df["performance"]=df["final_exam_score"].apply(
    lambda x:"Excellent" if x >= 85
    else "Good" if x >= 70
    else "Needs Improvement"
)

In [44]:
df.head()

,student_id,gender,study_hours,attendance,sleep_hours,parental_education,internet_access,extracurricular_activities,part_time_job,previous_grade,final_exam_score,final_grade,score_difference,performance
0,1,Male,4.0,98.0,6.5,Bachelors,Yes,Yes,No,76.9,100.0,A,23.1,Excellent
1,2,Female,6.3,100.0,5.7,High School,Yes,Yes,Yes,75.5,100.0,A,24.5,Excellent
2,3,Male,4.9,85.3,7.9,Bachelors,Yes,No,Yes,88.5,97.3,A,8.8,Excellent
3,4,Male,2.6,77.5,8.0,High School,Yes,Yes,No,85.1,83.8,B,-1.3,Good
4,5,Male,2.2,89.6,4.6,Bachelors,Yes,No,Yes,61.8,68.3,D,6.5,Needs Improvement


<h3>INSIGHT :</h3> Students were classified as Excellent, Good, or Needs Improvement. The classification is based on exam scores. It makes performance easier to understand.

In [45]:
df["final_grade"].value_counts()

final_grade
B    354
A    284
C    261
D     89
F     12
Name: count, dtype: int64

<h3>INSIGHT :</h3> Grade B is the most common grade with 354 students. Grade A is the second most common. Most students achieved good academic performance.


In [46]:
df.to_csv("student perfomance set.csv",index=False)

<h2><b><center> CONCLUSION</center></b></h2>

The analysis of the student performance dataset provides useful insights into the factors affecting student's academic performance. The analysis helps identify high and low performing students and understand different student characteristics. Study time shows a positive relationship with final exam scores, while attendance, sleep, internet access, and parental education also provide useful information about student performance. Most students have good academic performance, with Grade B being the most common.